twisted-infra/braid
# Move buildbot (master) submodule as a simple folder of braid repo.

Problem
=======

This is related to #79.

It should also help with #70 

From my point of view (as an outside / new contributor) braid is not a library but rather the core code which helps maintain Twisted infra. I also think that the usage of braid as a library is abused. It is ok to keep it in a separate package but not in separate repo.

In case braid became popular and is used outside of Twisted we can create a separate repo for it... but until then I think that we are overgineering this process.

Changes
=======

This is a minimal change.

I have done 2 commits. One the raw import  and another one changing the submodule and the way buildbot submodule is deployed.

buildbot fabric definition is still broken for vagrant/testing environment ... but I will fix it in a separate ticket once this is merged.


How to test
=========

Main changes are here https://github.com/twisted-infra/braid/commit/701f76d5e9412f8fe66bec9c102c5e3d675b0d7a

Since the definition is broken for the vagrant env... this can only be tested in production.

The diff is

brain repo in PARENT/braind
twisted-buildbot-configuration repo in PARENT/twisted-buildbot-configuration
Mare sure twisted-buildbot-configuration is on master branch with latest version,
Switch braid repo to this branch. Run a diff
```
meld services/buildbot/ ../twisted-buildbot-configuration
```

Check that files are the same... fab file is changed as it no longer get the repo but push the local files.

Thanks!
